3주차과제/cardGame/src/components/gameMain/GameMain.jsx

```diff
@@ -0,0 +1,167 @@
+      const newState = [...prevState];
+      newState[index] = !newState[index];
+      return newState;
+    });
+
+    setSelectedCards((prevSelectedCards) => [...prevSelectedCards, { item, index }]);
+
+    // 카드 비교
+    if (selectedCards.length === 1) {
+      const [firstCard] = selectedCards;
+      if (firstCard.item.name === item.name) {
+        setMatchedCards((prevMatchedCards) => [...prevMatchedCards, firstCard.item.name]);
+        setSelectedCards([]);
+      } else {
+        setTimeout(() => {
+          setIsFlipped((prevState) => {
+            const newState = [...prevState];
+            newState[firstCard.index] = false;
+            newState[index] = false;
+            return newState;
+          });
+          setSelectedCards([]);
+        }, 1000);
+      }
+    }
+  };
+
+  return (
+    <GameMainWrapper>
+      <GameLevelWrapper>
+        <GameLevelBtn onClick={() => onClickGameLevel(5)} $selected={gameLevel === 5}>
+          EASY
+        </GameLevelBtn>
+        <GameLevelBtn onClick={() => onClickGameLevel(7)} $selected={gameLevel === 7}>
+          NORMAL
+        </GameLevelBtn>
+        <GameLevelBtn onClick={() => onClickGameLevel(9)} $selected={gameLevel === 9}>
+          HARD
+        </GameLevelBtn>
+      </GameLevelWrapper>
+      <CardItemWrapper>
+        {shuffledCardItems.map((item, idx) => (
+          <CardItem
+            key={item.name + idx.toString()}
+            onClick={() => onClickCardItem(item, idx)}
+            className={isFlipped[idx] ? 'flipped' : ''}
+          >
+            <CardItemFront src={item.src} alt={item.name} className={isFlipped[idx] ? 'flipped' : ''} />
+            <CardItemBack />
+          </CardItem>
+        ))}
+      </CardItemWrapper>
+    </GameMainWrapper>
+  );
+};
+
+export default GameMain;
+
+const GameMainWrapper = styled.main`
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-start;
+  align-items: center;
+  gap: 3rem;
+
+  width: 95%;
+`;
+
+const GameLevelWrapper = styled.section`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  gap: 1.5rem;
+`;
+
+const GameLevelBtn = styled.button`
+  padding: 0.4rem 0.8rem;
+
+  border: 4px double ${({ theme, $selected }) => ($selected ? theme.colors.gray : theme.colors.black)};
+
+  background-color: ${({ theme, $selected }) => ($selected ? theme.colors.blue : theme.colors.gray)};
+  color: ${({ theme, $selected }) => ($selected ? theme.colors.white : theme.colors.black)};
+
+  font-size: 1.4rem;
+
+  cursor: pointer;
+`;
+
+const CardItemWrapper = styled.section`
+  display: flex;
+  justify-content: center;
+  flex-wrap: wrap;
+  gap: 1rem;
+
+  width: 100%;
+`;
+
+const CardItem = styled.div`
+  position: relative;
+
+  width: 13rem;
+  height: 15rem;
+
+  transition: 1s ease-in-out;
+  transform-style: preserve-3d;
+  perspective: 1000px;
+  transform: rotateY(${({ className }) => (className === 'flipped' ? '180deg' : '0deg')});
+`;
```

Comment:
오 카드가 뒤집혔는지 여부에 대한 `prop`을 `className`으로 받아오셨군요 굿입니다!
`className`대신 카드가 뒤집혀졌는지 여부를 boolean으로 표현하여 `isFlipped`으로 고쳐준다면 다른 사람이 봤을 때 카드가 뒤집혔는지 여부를 나타낸 부분인지 파악하기 쉬울것 같아요~ 

그리고 css 순서에 따라서 줄바꿈을 해주셔서 보기 더 편하네요 굿굿~

```suggestion


  const CardItem = styled.div`
    position: relative;
    
    width: 13rem;
    height: 15rem;
    
    transition: 1s ease-in-out;
    transform-style: preserve-3d;
    perspective: 1000px;
    transform: rotateY(${({ $isFilipped }) => ( $isFilipped ? '180deg' : '0deg')});
  `;


```